mypy/semanal.py

```diff
@@ -1483,7 +1486,8 @@ def visit_import_all(self, i: ImportAll) -> None:
             self.add_submodules_to_parent_modules(i_id, True)
             for name, node in m.names.items():
                 node = self.normalize_type_alias(node, i)
-                if not name.startswith('_') and node.module_public:
```

Comment:
Why isn't it possible to fix this by _only_ changing the logic here? For example add a nested ``if`` that ignores the check for underscore here if ``m.names`` contains ``'__all__'`` TBH, I don't really like inclusion of a new flag.